components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts

```diff
@@ -1053,8 +1050,7 @@ export const useSwap = () => {
 
     if (quoteUnion.lifiQuote) {
       const route = quoteUnion.lifiQuote.routes.find(
-        (route) =>
-          getUniqueRouteId('Li.Fi', route.steps) === selectedQuoteOptionId
+        (route) => route.uniqueId === selectedQuoteOptionId
       )
```

Comment:
We need a `fallback` here as well or else no route will be found if `selectedQuoteOptionId` is `undefined`.

```suggestion
      const route =
        quoteUnion.lifiQuote.routes.find(
          (route) => route.uniqueId === selectedQuoteOptionId
        ) || quoteUnion.lifiQuote.routes[0]
```

Reply:
Fixed here: [aa3fb85](https://github.com/brave/brave-core/pull/24961/commits/aa3fb85c2084eb3e5c7cd314eb229424f04a447b)